src/Plugin/views/display/Rules.php

```diff
@@ -0,0 +1,143 @@
+<?php
+namespace Drupal\rules\Plugin\views\display;
+
+use Drupal\rules\Context\ContextDefinition;
+use Drupal\views\Annotation\ViewsDisplay;
+use Drupal\views\Plugin\views\display\DisplayPluginBase;
+
+/**
+ * @ViewsDisplay(
+ *   id = "rules",
+ *   title = @Translation("Rules"),
+ *   admin = @Translation("Rules entity source"),
+ *   help = @Translation("Provide views results to rules workflows."),
+ *   theme = "views_view",
+ *   register_theme = FALSE,
+ *   uses_menu_links = FALSE,
+ *   rules = TRUE
+ * )
+ */
+class Rules extends DisplayPluginBase {
```

Comment:
Let's call this "Rules context provider" ? Theoretically this could be used by other modules also.